### PR TITLE
FEATURE: Add Infoblox NIOS DNS provider

### PR DESCRIPTION
**Pull request by matthewmgamble**
## Summary

- Adds a new DNS provider for [Infoblox NIOS](https://www.infoblox.com/) DDI appliances via the WAPI (Web API)
- Supports A, AAAA, CAA, CNAME, MX, PTR, SRV, and TXT record types
- Includes full provider documentation, integration test profile, and CI configuration

## Details

The provider connects to the Infoblox Grid Master using WAPI with HTTP Basic Auth and manages DNS records in a configurable DNS view. Key implementation notes:

- HTTP compression is disabled to avoid 403 responses from the Infoblox Apache reverse proxy
- Non-standard WAPI fields (CAA, PTR) are explicitly requested via `_return_fields+`
- NS records are excluded because Infoblox requires an `addresses` field (nameserver IPs) that DNSControl does not provide
- TXT records are limited to 255 bytes per Infoblox's single-string constraint

## Integration test results

Tested against Infoblox NIOS WAPI v2.12. Out of ~113 integration tests:
- **110 pass**
- 3 known failures due to Infoblox limitations (NS delegation requires `addresses` array, 256-byte TXT edge case)

## Files changed

**Provider implementation** (from prior commit):
- `providers/infoblox/` — provider code, API client, type converters, audit rules, unit tests

**WAPI compatibility fixes:**
- `providers/infoblox/api.go` — DisableCompression, per-type return field handling
- `providers/infoblox/convert.go` — Remove NS from supported types, strip TTL from NS bodies
- `providers/infoblox/auditrecords.go` — Reject empty TXT, tighten to 255-byte limit

**Submission files:**
- `README.md` — Added Infoblox to provider table
- `.github/CODEOWNERS` — `providers/infoblox @matthewmgamble`
- `.github/workflows/pr_integration_tests.yml` — INFOBLOX env vars
- `documentation/provider/infoblox.md` — Provider documentation
- `documentation/SUMMARY.md`, `documentation/provider/index.md` — Updated indexes
- `.github/labeler.yml`, `.goreleaser.yml` — Auto-generated updates
- `integrationTest/profiles.json` — INFOBLOX test profile

## Test plan

- [x] Unit tests pass (`go test ./providers/infoblox/...`)
- [x] `go vet ./providers/infoblox/...` clean
- [x] `go generate ./...` and `bin/generate-all.sh` run without errors
- [x] Integration tests run against live Infoblox NIOS server (110/113 pass)

Please create the GitHub label `provider-INFOBLOX`.

### CODE REVIEW

File: providers/infoblox/auditrecords.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+// AuditRecords returns a list of errors corresponding to the records
+// that aren't supported by this provider. If all records are
+// supported, an empty list is returned.
+func AuditRecords(records []*models.RecordConfig) []error {
+	a := rejectif.Auditor{}
+
+	a.Add("MX", rejectif.MxNull)
+
```
Comment (TomOnTime) [inline providers/infoblox/auditrecords.go:15]: Please add something that rejects NS records

File: documentation/provider/infoblox.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+- TXT
+
+## Limitations
+
+- **NS records are not supported.** Infoblox requires an `addresses` field (nameserver IPs) when creating NS delegation records, which DNSControl does not provide.
+- **Zones must be pre-created.** DNSControl cannot create zones in Infoblox; they must already exist in the configured DNS view.
+- **TXT records are limited to 255 bytes.** Infoblox does not support multi-string TXT records or single strings longer than 255 bytes.
+- **Empty TXT records are rejected.**
+- **TXT records with backslashes are not supported.**
+- **Concurrent operations are not supported.** The provider processes changes sequentially.
```
Comment (TomOnTime) [inline documentation/provider/infoblox.md:76]: FYI: Concurrent operations only downloads zone's records in parallel. The updates are done sequentially.  I'd be surprised if concurrent operations didn't work. Did you try it and see?

File: providers/infoblox/infobloxProvider.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+var features = providers.DocumentationNotes{
+	// The default for unlisted capabilities is 'Cannot'.
+	// See providers/capabilities.go for the entire list of capabilities.
+	providers.CanConcur:              providers.Cannot(),
+	providers.CanGetZones:            providers.Cannot(), // MVP: no ListZones
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.DocCreateDomains:       providers.Cannot("Infoblox zones must be pre-created"),
```
Comment (TomOnTime) [inline providers/infoblox/infobloxProvider.go]: No need to repeat the name of the provider

```suggestion
	providers.DocCreateDomains:       providers.Cannot("zones must be pre-created"),
```

File: providers/infoblox/convert.go [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+	rc := &models.RecordConfig{
+		Type:     "CNAME",
+		TTL:      effectiveTTL(r.UseTTL, r.TTL, defaultTTL),
+		Original: r.Ref,
+	}
+	rc.SetLabelFromFQDN(r.Name, domain)
+	if err := rc.PopulateFromString("CNAME", ensureTrailingDot(r.Canonical), domain); err != nil {
+		return nil, fmt.Errorf("failed to populate CNAME record: %w", err)
+	}
```
Comment (TomOnTime) [inline providers/infoblox/convert.go:173]: PopulateFromString() went away in v5.0, replaced by the much more powerful dc.NewRecordConfigParse/dc.NewRecordConfig.

This should be something like:

rc := dc.NewRecordConfig(dc.LabelFromShort(r.Name), effectiveTTL(r.UseTTL, r.TTL, defaultTTL), dnsv2.TypeCNAME, r.Canonical)
rc.Original = r.Ref
